integration_tests/src/main/python/cache_test.py

```diff
@@ -61,7 +62,7 @@ def test_passing_gpuExpr_as_Expr():
 @pytest.mark.parametrize('join_type', ['Left', 'Right', 'Inner', 'LeftSemi', 'LeftAnti'], ids=idfn)
 @ignore_order
 def test_cache_join(data_gen, join_type):
-    if data_gen.data_type == BooleanType():
+    if spark_version() == "3.0.0" and data_gen.data_type == BooleanType():
         pytest.xfail("https://github.com/NVIDIA/spark-rapids/issues/350")
```

Comment:
I would point to the spark issue for all of these instead

https://issues.apache.org/jira/browse/SPARK-32672